src/source/StreamInfoProvider.c

```diff
@@ -49,31 +70,59 @@ STATUS freeStreamInfoProvider(PStreamInfo* ppStreamInfo)
     LEAVES();
     return retStatus;
 }
-
-// Creates H264 track info for video
-STATUS createH264VideoTrackInfo(PTrackInfo pTrackInfo)
+// Creates track info for video with given codec and sets the content type as per the codec
+STATUS createVideoTrackInfo(VIDEO_CODEC_ID videoCodecId, PCHAR contentType, PTrackInfo pTrackInfo)
 {
     STATUS retStatus = STATUS_SUCCESS;
 
     pTrackInfo->trackId = DEFAULT_VIDEO_TRACK_ID;
     pTrackInfo->codecPrivateData = NULL;
     pTrackInfo->codecPrivateDataSize = 0;
-    STRCPY(pTrackInfo->codecId, MKV_H264_AVC_CODEC_ID);
+    switch (videoCodecId) {
+        case VIDEO_CODEC_ID_H264:
+            STRCPY(pTrackInfo->codecId, MKV_H264_AVC_CODEC_ID);
+            STRCAT(contentType, MKV_H264_CONTENT_TYPE);
+            break;
+        case VIDEO_CODEC_ID_H265:
+            STRCPY(pTrackInfo->codecId, MKV_H265_HEVC_CODEC_ID);
+            STRCAT(contentType, MKV_H265_CONTENT_TYPE);
+            break;
+        default:
+            STRCPY(pTrackInfo->codecId, MKV_H264_AVC_CODEC_ID);
+            STRCAT(contentType, MKV_H265_CONTENT_TYPE);
+    }
     STRCPY(pTrackInfo->trackName, DEFAULT_VIDEO_TRACK_NAME);
     pTrackInfo->trackType = MKV_TRACK_INFO_TYPE_VIDEO;
 
     return retStatus;
 }
 
-// Creates AAC track info for audio
-STATUS createAacAudioTrackInfo(PTrackInfo pTrackInfo)
+// Creates track info for audio with given codec and sets the content type as per the codec
+STATUS createAudioTrackInfo(AUDIO_CODEC_ID audioCodecId, PCHAR contentType, PTrackInfo pTrackInfo)
 {
     STATUS retStatus = STATUS_SUCCESS;
 
     pTrackInfo->trackId = DEFAULT_AUDIO_TRACK_ID;
     pTrackInfo->codecPrivateData = NULL;
     pTrackInfo->codecPrivateDataSize = 0;
-    STRCPY(pTrackInfo->codecId, MKV_AAC_CODEC_ID);
+
```

Comment:
Let us add a NULL check for `contentType` and `pTrackInfo`. Something like:


`CHK(contentType != NULL, STATUS_NULL_ARG)`. Similarly for `pTrackInfo`

Reply:
Added the `CHK(pTrackInfo != NULL && pStreamInfo->streamCaps.contentType != NULL, STATUS_NULL_ARG);` to `createAudioVideoStreamInfo` and `createVideoStreamInfo` which call this function